src/library/scala/collection/immutable/Range.scala

```diff
@@ -165,6 +165,49 @@ sealed abstract class Range(
     }
   }
 
+  override final def indexOf[@sp(Int) B >: Int](elem: B, from: Int = 0): Int =
+    elem match {
+      case i: Int =>
+        val pos = posOf(i)
+        if (pos >= from) pos else -1
+      case _ => super.indexOf(elem, from)
+    }
+
+  override final def lastIndexOf[@sp(Int) B >: Int](elem: B, end: Int = length - 1): Int =
+    elem match {
+      case i: Int =>
+        val pos = posOf(i)
+        if (pos <= end) pos else -1
+      case _ => super.lastIndexOf(elem, end)
+    }
+
+  private[this] def posOf(i: Int): Int =
+    if (contains(i)) (i - start) / step else -1
+
+  override def sorted[B >: Int](implicit ord: Ordering[B]): IndexedSeq[Int] =
+    if (ord eq Ordering.Int) {
+      if (step > 0) this // low-to-high already
+      else if (isEmpty) this // no elements are already sorted
+      else Range.inclusive(start = last, end = start, step = -step)
+    } else super.sorted(ord)
+
+  override def sameElements[B >: Int](that: IterableOnce[B]): Boolean = that match {
+    case other: Range =>
+      if (this.isEmpty && other.isEmpty) true
+      else {
+        (this.length : @annotation.switch) match {
+          case 0 => other.isEmpty
+          case 1 => other.length == 1 && this.start == other.start
+          case n => other.length == n && (
+            (this.start == other.start)
```

Comment:
what about reverse ordered Range?
Range.inclusive(1,10,1) == Range.inclusive(10,1,-1)

Reply:
no, they're unequal; `sameElements` is confusingly named in that it means "same elements and in the same order"